### PR TITLE
interchange_capnp: fix enum assignment when reading phys netlist

### PR DESCRIPTION
**Pull request by acomodi**
Signed-off-by: Alessandro Comodi <acomodi@antmicro.com>

### CODE REVIEW

File: fpga_interchange/interchange_capnp.py [hunk 1]
```diff
@@ -729,8 +729,8 @@ def to_physical_netlist(phys_netlist_capnp):
 
     for physical_cell in phys_netlist_capnp.physCells:
         phys_netlist.add_physical_cell(
-            strs[physical_cell.cellName],
-            PhysicalCellType(physical_cell.physType))
+            strs[physical_cell.cellName], PhysicalCellType[first_upper(
+                str(physical_cell.physType))])
```
Comment (litghost) [inline fpga_interchange/interchange_capnp.py:733]: Is the `str` here required?  I don't believe so?

**Review by litghost — CHANGES_REQUESTED** [review]
Please extend a test to check this change.

File: fpga_interchange/interchange_capnp.py (hunk 1 above)
Comment (acomodi) [reply fpga_interchange/interchange_capnp.py:733]: Without that I get a:
```
TypeError: 'capnp.lib.capnp._DynamicEnum' object is not subscriptable
```

File: tests/example_netlist.py [hunk 2]
```diff
@@ -347,6 +347,13 @@ def main():
     with open(args.physical_netlist, 'wb') as f:
         write_capnp_file(phys_netlist_capnp, f)
 
+    # Test reading back logical and physical netlists
```
Comment (litghost) [inline tests/example_netlist.py:350]: This isn't actually tested in the CI, this doesn't do what you want.

Comment (litghost) [reply tests/example_netlist.py:350]: See https://github.com/SymbiFlow/python-fpga-interchange/blob/b26bd256f25b9028a7240189042f4c131f143ee6/tests/test_basics.py#L25

Comment (acomodi) [reply tests/example_netlist.py:350]: Oh right, I see that now, gonna fix it right away

File: tests/example_netlist.py [hunk 3]
```diff
@@ -314,6 +314,9 @@ def example_physical_netlist():
         stubs=[],
         net_type=PhysicalNetType.Vcc)
 
+    phys_netlist.add_physical_cell(cell_name='ibuf', cell_type='port')
+    phys_netlist.add_physical_cell(cell_name='obuf', cell_type='port')
```
Comment (litghost) [inline tests/example_netlist.py:318]: cell_type should be the physical_netlist.PhysicalCellType enum, not a string.

Comment (acomodi) [reply tests/example_netlist.py]: Right, fixed

**Pull request merged**